### DISCUSSION TITLE
16 Bit PNG Import (Proposal)

### DISCUSSION THREAD

**Discussion by petterthowsen** [godotengine/godot-proposals#4143] [General Discussions]
I've been implementing a open world terrain system, and have been struggling to get 16 bit heightmap data imported into godot.

I'm confused as to why 16 bit-per-channel images are not supported, given their common use in heightmaps.

Having the ability to select "high resolution" in the import settings for the resource would be great and definitively solve some headache for me at least.

**Comment by fire** [comment]
https://github.com/godotengine/godot/pull/19391#issuecomment-412178102 was rejected a few years ago.

I recommend you use openexr.

**Reply by petterthowsen** [reply]
Hm.. ok, I'll see if I can find a program to batch convert the pngs to exr then 🤔

**Reply by Calinou** [reply]
To clarify, the original PR would probably be good to merge if it was an opt-in import option. There was no way to opt out of the behavior, which made the original PR problematic.

For reference, this was also posted on the Godot forums: https://godotforums.org/discussion/29201/how-to-import-16-bit-per-channel-pngs

**Comment by petterthowsen** [comment]
Ok, so... I decided to try to use OpenEXR. Getting my 16384 png files converted to OpenEXR is a pain, because the only program I've found that reliably converts them is Affinity Photo which isn't free (I have a trial version).

Even so, it adds another ~15 minute step for the conversion which is annoying.

We need 16 bit PNG support in Godot, here's why:

Support for EXR is lacking everywhere.

- Author of TerreSculptor has EXR on the list of todo's but is unlikely to implement it in the near future.
- The common Imagick image tool only has EXR support when compiled from source with extra libraries, which takes a lot of time and requires many build tools installed.

A few other converters lack the ability to handle many files, even if they support EXR.

So, we should have a "optional" opt-in support for 16 bit PNG when needed.

Thank you.

**Comment by shuvit-game** [comment]
I have spent a few months blindly working through this, I just wanted to share my current workflow for anyone searching this issue.

Gimp > 2.10 can read and write exr's fine. One pc I had was on 2.8 and it took me a while to figure this out.

I have a terrain generator that exports exr from godot, I can import it into blender, displace, sculpt, bake and export a 16bit png, use gimp to convert it to exr to send it back to godot. Of course there is loss going from 32bit to 16 and back, but so far this is giving me the detail I need, and two way communication between blender and godot.

However, strong +1 for native 16bit png support.

**Reply by fire** [reply]
The openexr format supports 32bit floating point. Not sure if godot engine tinyexr can open.

**Reply by shuvit-game** [reply]
Edit: my mistake!! I didn't realize blender could export an exr. Sorry!!

-However, Blender baking is limited to a 16 bit png.-
